sql/sql_table.cc

```diff
@@ -11148,6 +11158,42 @@ bool check_engine(THD *thd, const char *db_name,
   DBUG_RETURN(false);
 }
 
+bool add_hidden_vfield_fields(THD *thd, List<Create_field> *create_list,
+                              List<Key> *key_list, Alter_info *alter_info)
+{
+  if (alter_info->is_hidden_fields_processed)
+    return true;
+
+  List_iterator<Key> key_it(*key_list);
+  Key *key;
+  Key_part_spec *part;
+  while ((key = key_it++))
+  {
+    List_iterator<Key_part_spec> part_it(key->columns);
+    while ((part = part_it++))
+    {
+      if (!part->vfield)
+        continue;
+
+      Create_field *fld = mysql_add_invisible_field(thd,
+                                                    create_list,
+                                                    "hidden_vfield_expr",
+                                                    part->vfield->expr
+                                                                ->type_handler(),
+                                                    INVISIBLE_USER,
```

Comment:
INVISIBLE_SYSTEM, or INVISIBLE_FULL. This will fix the issue with SHOW CREATE, and eliminate the need of force_gen_name parameter as well